### DISCUSSION TITLE
Making use of various LEDs on the SL49? Making a decent mixing preset for SL49

### DISCUSSION THREAD

**Discussion by ObservantSound** [helgoboss/helgobox#1387] [Q&A]
Hi. I'm working on making my custom mixing controler with your fantastic script.
Using my faders and pots for relative track selection is already great! 

I've been struggling to come up with a good solo/mute system though...
The supported partial HUI is very lacking and not fun to work with in my opinion.

This is what I've tried:
1. Source: Midi Note --> Target: track/solo
2. Source: Midi Feedback --> Target: track/solo

Doing this will light up a keys led strip according to the selected tracks solo state. So all as intended.
I would prefer though to do this with either the pads or buttons but have not managed to do so.
If I use a pad/button with either cc or a midi number as source then it triggers correctly, but the LED feedback only works if the source is also a midi note.
Setting button/pad to toggle in realearn does not reflect the light status on the SL49.
Setting the SL49 button/pads to toggle, while choosing "normal" mode in realearn also works in the sense, that if the button/pad is turned on but the track is not soloed, then pressing the button again kind of acts like a catch up to be in sync again with the track solo state. 
But changing a track, soloing something with the mouse or other means does not reflect.

Is this a hardware/software limitation or is there a way to achieve this?
I'd be happy to make a bigger preset and make it available once it's ready.

**Comment by helgoboss** [comment]
Do you mean Novation 49SL Mk3? Either way, I don't have it, so I can't speak about hardware limitations.

So everything from here is about the software part.

If there's no (or no good) controller preset for your device already, you need to know exactly which MIDI messages your device expects to light which LEDs. Without that knowledge, it's trial and error.

When you have figured that out, it's sometimes good to separate your mappings into control-only and feedback-only mappings. E.g. if you want to control something with a button but want the feedback to appear somewhere else, you need to use 2 mappings, one control mapping (with feedback disabled) and one feedback mapping (with control disabled). With that principle alone, you can model many common control scenarios.

If this not enough info, send some pictures or a video to describe what you want to do.

**Comment by ObservantSound** [comment]
Hey. Sorry for the late reply. It kind of fell off my radar.
So I think what I was trying to do just doesn't work with the 49SL MK3.

The top right buttons don't seem to respond to feedback and only change their light state by pushing them.
The middle pads only light up when being used with the sequencer and only send out notes/cc etc. in grid mode, where the LEDs don't react to touch.
The transport buttons also don't seem to send out any messages.

It does have a DAW HUI mode, but it's quite limiting.
So unless I'm doing something wrong, I think it just doesn't work.
Which is fine :) 
There's still endless other possibilities
